vault/identity_store_oidc.go

```diff
@@ -1685,11 +1687,20 @@ func (i *IdentityStore) generatePublicJWKS(ctx context.Context, s logical.Storag
 		}
 
 		for _, keyID := range keyIDs {
-			key, err := loadOIDCPublicKey(ctx, s, keyID)
-			if err != nil {
-				return nil, err
+			found := false
+			for _, key := range jwks.Keys {
+				if key.KeyID == keyID {
+					found = true
```

Comment:
Do we want to `break` if it matches, and then `continue` (outside of this loop) if found?

An alternative to looping over `jwks.Keys` would be to call [`strutil.RemoveDuplicates`](https://github.com/hashicorp/go-secure-stdlib/blob/main/strutil/strutil.go#L232) against it at the very end.

Reply:
Took different approach in e7e2858.